features/step_definitions/migration/migration_steps.rb

```diff
@@ -0,0 +1,128 @@
+Given(/^At least (\d+) Xen Hypervisors are available$/) do |arg1|
+  # get hypervisors and count Xen's ones - for migration at least 2
+  hashypervisor_xen = 0
+  @hypervisors = control_node.openstack.hypervisor.list
+  expect(@hypervisors).not_to be_empty
+
+  @hypervisors.each do |h|
+    hypervisor = control_node.openstack.hypervisor.show(h.id)
+    hashypervisor_xen = (hashypervisor_xen + 1) if hypervisor.hypervisor_type == "Xen"
+  end
```

Comment:
In Ruby you have nice methods for handling collections, you could use `select` here:
```ruby
xen_nodes = @hypervisors.select do |hv|  
  control_node.openstack.hypervisor.show(hv.id).hypervisor_type == "Xen"
end
expect(xen_node.size).to be >= arg1.to_i
```